### DISCUSSION TITLE
How to replace the path for fetching pyodide packages with another mirror source

### DISCUSSION THREAD

**Discussion by JackYinpei** [pyodide/pyodide#4283] [Q&A]
hello

I used pyodide in the jupyterlite project, and used micropip to load some uncommon packages. These packages are hosted under the pypi.org/simple path.

As I said in the title, I want to change the path where micropip downloads packages from pypi.org to another path, because accessing resources on pypi.org from our country is too slow.

![image](https://github.com/pyodide/pyodide/assets/49555788/5e1d1589-d53b-4acd-b0d2-1a5a2fa38e56)

I would greatly appreciate any feedback or suggestions you can provide. Thanks in advance

**Comment by rth** [comment]
You can specify a custom index with [`micropip.set_index_urls`](https://micropip.pyodide.org/en/stable/project/api.html#micropip.set_index_urls) which supports  Simple Repository Index (PEP503). However, I think using JupyterLite's `%pip install` would still work as it passes the arguments to micropip, but worth double checking. 

Let me know if it doesn't work

**Reply by JackYinpei** [reply]
In fact, over the past few days I have tried recompiling pyodide and replacing the default micropip address with a mirror address. 

However, this did not work successfully. 

Now I want to find the steps taken by jupyterlite or pyodide before installing the comm package, and use `micropip.set_index_urls` before installing this package. But I still have not found it..
